test_reporting/junit_xml_parser.py

```diff
@@ -342,6 +342,9 @@ def parse_test_result(roots):
         A dict containing the parsed test result.
     """
     test_result_json = defaultdict(dict)
+    if not roots:
+        print("The pipeline failed, no XML file needs to parse.")
+        return
```

Comment:
This will return None and may cause other code dependent on this function to raise exception. I think we need a better way to handle unexpected situation. Like raise an exception explicitly. Then other code can handle the exception in an elegant way.

Reply:
@wangxin Thank you for pointing this out. Updated.